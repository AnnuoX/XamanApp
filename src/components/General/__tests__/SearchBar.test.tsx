import React from 'react';
// Note: test renderer must be required after react-native.

import renderer from 'react-test-renderer';

import { SearchBar } from '..';

jest.useFakeTimers();

describe('[SearchBar]', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<SearchBar />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should show clear button on text enter', () => {
        const tree = renderer.create(<SearchBar />);
        const { instance } = tree.root;
        instance.onChangeText('searched text');

        expect(instance.state.value).toBe('searched text');

        jest.advanceTimersByTime(1000);

        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should clear text', () => {
        const tree = renderer.create(<SearchBar />);
        const { instance } = tree.root;

        instance.onClearPress();

        jest.advanceTimersByTime(1000);

        expect(instance.state.value).toBe('');
        expect(tree.toJSON()).toMatchSnapshot();
    });
});
