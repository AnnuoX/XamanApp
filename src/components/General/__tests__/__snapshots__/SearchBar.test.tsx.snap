// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SearchBar] renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F3F6FA",
        "borderRadius": 15,
        "flexDirection": "row",
        "height": 73.5,
      },
      false,
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flexDirection": "column",
        "left": 0,
        "position": "absolute",
        "tintColor": "#000000",
        "width": 50,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "IconSearch",
        }
      }
      style={
        Array [
          Object {
            "height": 31.428571428571427,
            "width": 31.428571428571427,
          },
          Object {
            "tintColor": "#000000",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Please type ..."
      placeholderTextColor="#606885"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "UbuntuMono-Bold",
            "fontSize": 34.285714285714285,
            "fontWeight": "600",
            "height": "100%",
            "paddingLeft": 50,
            "paddingRight": 50,
            "paddingVertical": 0,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "tintColor": "#000000",
        "width": 50,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "alignSelf": "center",
          "opacity": 0,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "IconX",
          }
        }
        style={
          Array [
            Object {
              "height": 31.428571428571427,
              "width": 31.428571428571427,
            },
            Object {
              "tintColor": "#000000",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`[SearchBar] should clear text 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F3F6FA",
        "borderRadius": 15,
        "flexDirection": "row",
        "height": 73.5,
      },
      false,
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flexDirection": "column",
        "left": 0,
        "position": "absolute",
        "tintColor": "#000000",
        "width": 50,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "IconSearch",
        }
      }
      style={
        Array [
          Object {
            "height": 31.428571428571427,
            "width": 31.428571428571427,
          },
          Object {
            "tintColor": "#000000",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Please type ..."
      placeholderTextColor="#606885"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "UbuntuMono-Bold",
            "fontSize": 34.285714285714285,
            "fontWeight": "600",
            "height": "100%",
            "paddingLeft": 50,
            "paddingRight": 50,
            "paddingVertical": 0,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "tintColor": "#000000",
        "width": 50,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "alignSelf": "center",
          "opacity": 0,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "IconX",
          }
        }
        style={
          Array [
            Object {
              "height": 31.428571428571427,
              "width": 31.428571428571427,
            },
            Object {
              "tintColor": "#000000",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`[SearchBar] should show clear button on text enter 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F3F6FA",
        "borderRadius": 15,
        "flexDirection": "row",
        "height": 73.5,
      },
      false,
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flexDirection": "column",
        "left": 0,
        "position": "absolute",
        "tintColor": "#000000",
        "width": 50,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "IconSearch",
        }
      }
      style={
        Array [
          Object {
            "height": 31.428571428571427,
            "width": 31.428571428571427,
          },
          Object {
            "tintColor": "#000000",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Please type ..."
      placeholderTextColor="#606885"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "UbuntuMono-Bold",
            "fontSize": 34.285714285714285,
            "fontWeight": "600",
            "height": "100%",
            "paddingLeft": 50,
            "paddingRight": 50,
            "paddingVertical": 0,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value="searched text"
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "tintColor": "#000000",
        "width": 50,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "alignSelf": "center",
          "opacity": 1,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "IconX",
          }
        }
        style={
          Array [
            Object {
              "height": 31.428571428571427,
              "width": 31.428571428571427,
            },
            Object {
              "tintColor": "#000000",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
